woocommerce/woocommerce-ios
# Order Details > Shipping Label: reprint shipping label main screen

Part of #2169 

For code reviewers: please review https://github.com/woocommerce/woocommerce-ios/pull/3341 first. Feel free to hold off reviewing this PR if the base PR has changes requested.

## Why

This PR implements the main screen for reprinting a shipping label. The view model is implemented in the base PR https://github.com/woocommerce/woocommerce-ios/pull/3341 since the UI changes are large enough.

## Changes

- Added a new icon `pagesFootnoteImage` for the footnote size for `pages` Gridicon
- Added an extension `UIButton.applyFootnoteLinkButtonStyle` tot apply footnote link button styles (footnote has smaller font size)
- Added an extension `UITableViewCell.hideSeparator` to hide the separator for a cell - feel free to suggest other solutions!
- Created `ReprintShippingLabelViewController` with a table view and a CTA to reprint a shipping label. I decided to implement the UI above the CTA with table view cells, after struggling with dynamic height using `tableView.tableHeaderView`.)
- Created `SpacerTableViewCell` for a cell just for vertical padding
- Integration with order details in `OrderDetailsViewController` and `OrderDetailsDataSource`

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/) and there is an order with at least one non-refunded shipping label.

You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

- Go to the orders tab
- Tap on an order with at least one non-refunded shipping label
- After syncing, tap on "Reprint Shipping Label" CTA --> it should navigate to the reprint UI as in screenshots

## Example screenshots

Notes for reviewers: due to the current implementation of `TopLeftImageTableViewCell` still using frames and bounds, the images in "See layout and paper sizes options" and "Don't know how to print from your device?" rows are not the footnote size 20x20 as in design. I added this to the design enhancements issue https://github.com/woocommerce/woocommerce-ios/issues/3334 to work on them separately.

light | dark
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-12-14 at 13 58 18](https://user-images.githubusercontent.com/1945542/102047199-cf6b0780-3e17-11eb-869c-8a0f1cecfb1c.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-14 at 14 24 57](https://user-images.githubusercontent.com/1945542/102047401-2cff5400-3e18-11eb-8dfe-46db86007fc9.png)



Update release notes:
